db/fixtures/ae_datastore/ManageIQ/Cloud/VM/Retirement/StateMachines/Methods.class/__methods__/amazon_pre_retirement.rb

```diff
@@ -4,12 +4,26 @@
 #
 
 vm = $evm.root['vm']
-if vm && vm.attributes['power_state'] == 'on'
-  ems = vm.ext_management_system
+ems = vm.ext_management_system if vm
+
+if vm.nil? || ems.nil?
+  $evm.log('info', "Skipping Amazon pre retirement for Instance:<#{vm.try(:name)}> on EMS:<#{ems.try(:name)}>")
+  exit MIQ_OK
+end
+
+power_state = vm.power_state
+if power_state == "on"
+  if vm.hardware.root_device_type.blank?
+    $evm.log('error', "Aborting Amazon pre retirement, empty root_device_type."\
+                      "Instance may have been provisioned outside of CFME <#{vm.name}>")
+    exit MIQ_ABORT
+  end
+
   if vm.hardware.root_device_type == "ebs"
-    $evm.log('info', "Stopping Amazon Instance <#{vm.name}> in EMS <#{ems.try(:name)}>")
-    vm.stop if ems
+    $evm.log('info', "Stopping EBS Amazon Instance <#{vm.name}> in EMS <#{ems.name}>")
+    vm.stop
   else
-    $evm.log('info', "Skipping stopping of non EBS Amazon Instance <#{vm.name}> in EMS <#{ems.try(:name)}>")
+    $evm.log('info', "Skipping stopping of non EBS Amazon Instance <#{vm.name}> in EMS <#{ems.name}> "\
+              "with type <#{vm.hardware.root_device_type}>")
```

Comment:
@billfitzgerald0120 
can we change the "with type" to "with instance store type"